firedrake/matrix.py

```diff
@@ -96,6 +96,7 @@ def __init__(self, a, bcs, mat_type, *args, **kwargs):
         # sets self._a, self._bcs, and self._mat_type
         super(Matrix, self).__init__(a, bcs, mat_type)
         options_prefix = kwargs.pop("options_prefix")
+        kwargs["mat_type"] = mat_type
         self.M = op2.Mat(*args, **kwargs)
```

Comment:
You can just say `op2.Mat(*args, mat_type=mat_type, **kwargs)`

Reply:
Great, thanks!